src/main/resources/blockly/js/mcreator_blockly.js

```diff
@@ -81,6 +81,16 @@ function arrayToBlocklyDropDownArray(arrorig) {
     return retval;
 }
 
+function arrayToBlocklyDropDownArrayWithReadableNames(readablenames, arrorig) {

```

Comment:
Consider swapping function arguments as I think it makes more sense to provide keys first, then readable names